src/libirmin/commit.ml

```diff
@@ -49,8 +49,8 @@ module Make (I : Cstubs_inverted.INTERNAL) = struct
       (repo @-> hash @-> returning commit)
       (fun (type repo) repo hash ->
         with_repo' repo commit
-          (fun (module Store : Irmin.Generic_key.S with type repo = repo) repo
-          ->
+          (fun
+            (module Store : Irmin.Generic_key.S with type repo = repo) repo ->
```

Comment:
Nitpick: I don't think I like breaking after the `fun` keyword because it creates weird alignments between the arguments and the body, but the original code is problematic to indent nicely anyway (so it doesn't matter much)

Reply:
The `fun` has to break if the arguments are even bigger and the arrow is better there to avoid a line with only the arrow:

```ocaml
          (fun
            (module Store : Irmin.Generic_key.S with type repo = repo)
            (module Store : Irmin.Generic_key.S with type repo = repo) ->
            body)
```

I agree with weird alignment. More indentation ?
```ocaml
          (fun
              (module Store : Irmin.Generic_key.S with type repo = repo)
              (module Store : Irmin.Generic_key.S with type repo = repo) ->
            body)
```

Reply:
Agree! Though, that's not possible without huge diffs. I would like to discuss this with more people here: https://github.com/ocaml-ppx/ocamlformat/issues/2397